src/main.rs

```diff
@@ -81,16 +89,100 @@ enum Cmd {
         #[clap(long)]
         leave_debug_open: bool,
         /// Requires that a subset of keys are enabled (and only that subset).
-        /// This flag takes a bitmask (in the base of your choice as long as
-        /// it's not octal) where bit 0 maps to RoTK 0, 1 maps to 1, and so
-        /// forth. If keys corresponding to 1 bits are not enabled, and keys
-        /// corresponding to 0 bits are not *not* enabled, this will decline to
-        /// lock.
-        #[clap(long, value_parser = parse_int::parse::<u8>)]
-        require_key_enable_shape: Option<u8>,
+        /// This flag takes a 4 character string where 'E' represents
+        /// enabled, 'R' revoked, and 'I' invalid. Keys are specified
+        /// in order from 3 to 0. A string of EERR means keys 0 and 1
+        /// are revoked and keys 2 and 3 are enabled.
+        /// If the bits do not match this will decline to lock!
+        #[clap(long, value_parser = KeyStateTable::from_str)]
+        require_key_enable_shape: Option<KeyStateTable>,
     },
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+enum KeyState {
+    Invalid,
+    Enabled,
+    Revoked,
+}
+
+impl std::fmt::Display for KeyState {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            KeyState::Invalid => write!(f, "i"),
+            KeyState::Enabled => write!(f, "e"),
+            KeyState::Revoked => write!(f, "r"),
+        }
+    }
+}
+
+impl TryFrom<u32> for KeyState {
+    type Error = anyhow::Error;
+    fn try_from(v: u32) -> Result<Self, Self::Error> {
+        Ok(match v {
+            0b00 => Self::Invalid,
+            0b01 => Self::Enabled,
+            0b10 | 0b11 => Self::Revoked,
+            _ => bail!("invalid value {v}"),
+        })
+    }
+}
+
+impl TryFrom<char> for KeyState {
+    type Error = anyhow::Error;
+    fn try_from(v: char) -> Result<Self, Self::Error> {
+        Ok(match v {
+            'i' | 'I' => Self::Invalid,
+            'e' | 'E' => Self::Enabled,
+            'r' | 'R' => Self::Revoked,
+            _ => bail!("invalid value {v}"),
+        })
+    }
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+struct KeyStateTable([KeyState; 4]);
+
+impl std::str::FromStr for KeyStateTable {
+    type Err = anyhow::Error;
+
+    // The user has requested that we ensure the CFPA has a particular set of
```

Comment:
This comment is kinda in a weird place here; it seems like a docstring for `from_str`, but doesn't make sense in that context.

Reply:
I think I need to split up this comment